### PR TITLE
Bring across @cloudflare/kv-asset-handler

### PR DESCRIPTION
**Pull request by GregBrimble**
Adds [@cloudflare/kv-asset-handler](https://github.com/cloudflare/kv-asset-handler) to workers-sdk.

A slightly wonky #4315.

```sh
# /path/to/filter.sh
#!/bin/sh

mkdir -p packages/kv-asset-handler
rsync -avC --remove-source-files . ./packages/kv-asset-handler
```

```sh
# in kv-asset-handler
FILTER_BRANCH_SQUELCH_WARNING=1 git filter-branch --tree-filter '/path/to/filter.sh'
git filter-repo --tag-rename '':'@cloudflare/kv-asset-handler@' --force
git filter-repo --tag-rename '@cloudflare/kv-asset-handler@v':'@cloudflare/kv-asset-handler@' --force
```

```sh
# in workers-sdk
git remote add kv-asset-handler-remote /path/to/kv-asset-handler
git fetch kv-asset-handler-remote
git checkout -b kv-asset-handler
git cherry-pick ..kv-asset-handler-remote/main --strategy-option theirs -m 1
# then spam `git cherry-pick --skip`
```

image

The only interesting commits are these four:
- https://github.com/cloudflare/workers-sdk/pull/4782/commits/b97d7b1f28f546ef39ec29831beb08b653bf5f03
- https://github.com/cloudflare/workers-sdk/pull/4782/commits/dff495acf1eddff589234ce6188a251b173c9be1
- https://github.com/cloudflare/workers-sdk/pull/4782/commits/4df8de8da9486381724c71e2aced38bce7fcdf74
- https://github.com/cloudflare/workers-sdk/pull/4782/commits/c3faba192df5e973d4ff4fb7c5b4df8cee390b42

### CODE REVIEW

**Comment by Cherry** [comment]
Nice to see this!! Is there any way to bring over the commit history like in https://github.com/cloudflare/workers-sdk/pull/4315 - I feel there's a lot of nuance and history there that would be good to keep?

**Comment by GregBrimble** [comment]
Yup, absolutely can do. That's a good shout, thanks! :)

File: CODEOWNERS [hunk 1]
```diff
@@ -17,6 +17,9 @@
 # C3 ownership
 /packages/create-cloudflare/ @cloudflare/c3
 
+# kv-asset-handler ownership
+/packages/kv-asset-handler/ @cloudflare/developer-advocacy @kristianfreeman @rickyrobinett @lauragift21 @LoganGrasby @craigsdennis
```
Comment (petebacondarwin) [inline CODEOWNERS:21]: Github seems to think some of these users/teams don't exist.

**Review by petebacondarwin — APPROVED** [review]
LGTM.

I think this now opts kv-asset-handler into the Changesets based deployment process, right?

Also is there a PR in flight to "turn off" the old repository so that people don't accidentally deploy from there? And also to transfer and redirect issues to this repo?

**Comment by GregBrimble** [comment]
It does indeed. That was the primary motivation for it actually, since `@cloudflare/kv-asset-handler` is solely managed by `wrangler-publisher` in npm, but no human has the keys to that.

I have magical admin powers to merge [this PR](https://github.com/cloudflare/kv-asset-handler/pull/402) and archive the repo once this one is in.

I'll bring across all issues and label them with `kv-asset-handler`.

File: CODEOWNERS (hunk 1 above)
Comment (GregBrimble) [reply CODEOWNERS:21]: Yes, they need to be added to this repo. cc. @lrapoport-cf

**Pull request merged**